api/handlers/capturesource.go

```
@@ -31,46 +31,184 @@ LICENSE
+func FromCaptureSource(captureSource *model.CaptureSource, id int, format *api.Format) CaptureSourceResult {
+	var result CaptureSourceResult
+	if format.Requires("id") {
+		result.ID = &id
+	}
+	if format.Requires("name") {
+		result.Name = &captureSource.Name
+	}
+	if format.Requires("location") {
+		location := fmt.Sprintf("%f,%f", captureSource.Location.Lat, captureSource.Location.Lng)
+		result.Location = &location
+	}
+	if format.Requires("camera_hardware") {
+		result.CameraHardware = &captureSource.CameraHardware
+	}
+	if format.Requires("site_id") {
+		result.SiteID = captureSource.SiteID
+	}
+	return result
+}
+
+// GetCaptureSourcesQuery describes the URL query parameters required for the GetCaptureSources endpoint.
+type GetCaptureSourcesQuery struct {
+	Name     *string `query:"name"`     // Optional.
+	Location *string `query:"location"` // Optional.
+	api.LimitAndOffset
+	// TODO: Code could be simplified if api.Format could be embedded here. Testing shows it cannot, reason unknown.
+}
+
+// CreateCaptureSourceBody describes the JSON format required for the CreateCaptureSource endpoint.
+// ID is omitted because it is chosen automatically. All other fields are required.
+type CreateCaptureSourceBody struct {
+	Name           string `json:"name"`
+	Location       string `json:"location"`
+	CameraHardware string `json:"camera_hardware"`
+	SiteID         *int64 `json:"site_id"` // Optional.
+}
+
+// GetCaptureSourceByID gets a capture source when provided with an ID.
 func GetCaptureSourceByID(ctx *fiber.Ctx) error {
-	// TODO: implement handler
+	// Parse URL.
+	format := new(api.Format)
+
+	if err := ctx.QueryParser(format); err != nil {
+		return api.InvalidRequestURL(ctx)
+	}
 
-	// id, _ := ctx.ParamsInt("id", 1)
-	return ctx.JSON("TODO")
+	id, err := strconv.ParseInt(ctx.Params("id"), 10, 64)
+	if err != nil {
+		return api.InvalidRequestURL(ctx)
```

The information contained in err is lost here.